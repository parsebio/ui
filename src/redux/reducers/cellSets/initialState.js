const initialState = {
  properties: {},
  hierarchy: [],
  initialLoadPending: true,
  loading: false,
  error: false,
  updatingClustering: false,
  hidden: new Set(),
};

export default initialState;
