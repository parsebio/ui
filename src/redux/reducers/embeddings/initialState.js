/**
 * The store has the following format:
 * {
 *    embeddingType: {
 *      data: {xValues:[1,2], yValues:[5,3], cellIds:[0,3]},
 *      loading: false,
 *      error: false
 *    }
 * }
 */

const initialEmbeddingState = {
  data: {},
  loading: true,
  error: false,
};

const initialState = {};

export { initialEmbeddingState };
export default initialState;
