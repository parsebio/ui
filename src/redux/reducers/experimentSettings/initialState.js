const metaInitialState = {
  loading: false,
  completingStepError: false,
  loadingSettingsError: false,
  changedQCFilters: new Set(),
};

const initialState = {
  info: {
    experimentId: null,
    experimentName: null,
    sampleIds: [],
    pipelineVersion: null,
  },
  processing: {
    meta: metaInitialState,
  },
  originalProcessing: {},
};

export { metaInitialState };
export default initialState;
