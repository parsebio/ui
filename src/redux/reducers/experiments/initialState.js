const experimentTemplate = {
  name: null,
  description: 'Analysis description',
  id: null,
  createdAt: null,
  updatedAt: null,
  notifyByEmail: true,
  sampleIds: [],
  metadataKeys: [],
  pipelineVersion: 1,
  parentExperimentId: null,
  cellLevelMetadata: null,
  isSubsetted: false,
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
    activeExperimentId: null,
  },
  ids: [],
};

const METADATA_DEFAULT_VALUE = 'N.A.';

export default initialState;
export { experimentTemplate, METADATA_DEFAULT_VALUE };
