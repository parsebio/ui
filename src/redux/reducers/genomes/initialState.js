const initialState = {
  loading: false,
  error: false,
  public: {},
  custom: {},
};

export default initialState;
