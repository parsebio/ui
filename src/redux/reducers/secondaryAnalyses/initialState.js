const secondaryAnalysisTemplate = {
  name: null,
  id: null,
  createdAt: null,
  kit: null,
  chemistryVersion: null,
  numOfSamples: null,
  numOfSublibraries: null,
  files: { data: {} },
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
    activeSecondaryAnalysisId: null,
  },
  ids: [],
};

export { secondaryAnalysisTemplate };
export default initialState;
