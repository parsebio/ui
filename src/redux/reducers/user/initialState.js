const initialState = {
  current: null,
  isAuthenticated: false,
};

export default initialState;
