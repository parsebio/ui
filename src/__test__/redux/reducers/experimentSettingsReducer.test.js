import { enableMapSet } from 'immer';
import experimentSettingsReducer from 'redux/reducers/experimentSettings';
import initialState from 'redux/reducers/experimentSettings/initialState';
import generateExperimentSettingsMock from '__test__/test-utils/experimentSettings.mock';

import {
  EXPERIMENT_SETTINGS_NON_SAMPLE_FILTER_UPDATE,
  EXPERIMENT_SETTINGS_PIPELINE_VERSION_UPDATED,
  EXPERIMENT_SETTINGS_PROCESSING_CONFIG_LOADED,
  EXPERIMENT_SETTINGS_PROCESSING_ERROR,
  EXPERIMENT_SETTINGS_SAMPLE_FILTER_UPDATE,
  EXPERIMENT_SETTINGS_SET_QC_STEP_ENABLED,
} from 'redux/actionTypes/experimentSettings';

import errorTypes from 'redux/actions/experimentSettings/errorTypes';

enableMapSet();

const initialExperimentState = generateExperimentSettingsMock(['sample-KO']);

describe('experimentSettingsReducer', () => {
  it('Reduces identical state on unknown action', () => expect(
    experimentSettingsReducer(undefined, {
      action: 'well/this/is/not/a/valid/action',
      payload: {},
    }),
  ).toEqual(initialState));

  it('Sets up loading state on loading action', () => {
    const testData = 'test data';
    const newData = {
      test: testData,
    };

    const newState = experimentSettingsReducer({ ...initialState }, {
      payload: { data: newData },
      type: EXPERIMENT_SETTINGS_PROCESSING_CONFIG_LOADED,
    });

    expect(newState.processing.meta.loading).toEqual(false);
    expect(newState.processing.test).toEqual(testData);
  });

  it('Properly sets error on state', () => {
    const errorMessage = 'ERROR : Failed to load state';
    const newState = experimentSettingsReducer({ ...initialState }, {
      payload: { error: errorMessage, errorType: errorTypes.LOADING_PROCESSING_SETTINGS },
      type: EXPERIMENT_SETTINGS_PROCESSING_ERROR,
    });

    expect(newState.processing.meta.loading).toEqual(false);
    expect(newState.processing.meta.loadingSettingsError).toEqual(errorMessage);
  });

  it('Updates existing value properly', () => {
    const newState = experimentSettingsReducer(
      initialState,
      {
        type: EXPERIMENT_SETTINGS_NON_SAMPLE_FILTER_UPDATE,
        payload:
        {
          step: 'configureEmbedding',
          configChange: { embeddingSettings: { method: 'newMethod' } },
        },
      },
    );

    expect(newState.processing.configureEmbedding.embeddingSettings.method).toEqual('newMethod');
    expect(newState).toMatchSnapshot();
  });

  it('Adds new value properly', () => {
    const newState = experimentSettingsReducer(
      initialState,
      {
        type: EXPERIMENT_SETTINGS_NON_SAMPLE_FILTER_UPDATE,
        payload:
        {
          step: 'configureEmbedding',
          configChange: { embeddingSettings: { newProperty: 'property' } },
        },
      },
    );

    expect(newState.processing.configureEmbedding.embeddingSettings.newProperty).toEqual('property');
    expect(newState).toMatchSnapshot();
  });

  it('Adds new object properly', () => {
    const newState = experimentSettingsReducer(
      initialState,
      {
        type: EXPERIMENT_SETTINGS_NON_SAMPLE_FILTER_UPDATE,
        payload:
        {
          step: 'configureEmbedding',
          configChange: { embeddingSettings: { newProperty: { name: 'a', value: 'b' } } },
        },
      },
    );

    expect(newState.processing.configureEmbedding.embeddingSettings.newProperty).toEqual({ name: 'a', value: 'b' });
    expect(newState).toMatchSnapshot();
  });

  it('Updates sample settings properly', () => {
    const newState = experimentSettingsReducer(
      initialExperimentState,
      {
        type: EXPERIMENT_SETTINGS_SAMPLE_FILTER_UPDATE,
        payload:
        {
          step: 'cellSizeDistribution',
          sampleId: 'sample-KO',
          diff: { binStep: 400 },
        },
      },
    );

    const expectedCellSizeDistribution = {
      'sample-KO': {
        auto: true,
        enabled: true,
        filterSettings: {
          minCellSize: 10800,
          binStep: 400,
        },
        defaultFilterSettings: {
          minCellSize: 10800,
          binStep: 200,
        },
      },
    };

    // New entry is created for sample-KO with the new binStep while default value isn't changed
    expect(newState.processing.cellSizeDistribution).toEqual(expectedCellSizeDistribution);

    // Nothing else changes
    expect(newState).toMatchSnapshot();
  });

  it('Changing filter enabled property updates sample filter enabled property as well', () => {
    const newState = experimentSettingsReducer(
      initialExperimentState,
      {
        type: EXPERIMENT_SETTINGS_SET_QC_STEP_ENABLED,
        payload:
        {
          step: 'cellSizeDistribution',
          enabled: false,
        },
      },
    );

    const expectedCellSizeDistribution = {
      'sample-KO': {
        auto: true,
        enabled: false,
        filterSettings: {
          minCellSize: 10800,
          binStep: 200,
        },
        defaultFilterSettings: {
          minCellSize: 10800,
          binStep: 200,
        },
      },
    };

    // New entry is created for sample-KO with the new enabled changed to equal that in payload
    expect(newState.processing.cellSizeDistribution).toEqual(expectedCellSizeDistribution);

    // Nothing else changes
    expect(newState).toMatchSnapshot();
  });

  it('Correctly updates pipelineVersion', () => {
    const newState = experimentSettingsReducer(initialExperimentState, {
      type: EXPERIMENT_SETTINGS_PIPELINE_VERSION_UPDATED,
      payload: {
        pipelineVersion: 2,
      },
    });

    expect(newState.info.pipelineVersion).toEqual(2);
    expect(newState).toMatchSnapshot();
  });
});
