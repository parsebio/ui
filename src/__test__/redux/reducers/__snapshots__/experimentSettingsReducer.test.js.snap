// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`experimentSettingsReducer Adds new object properly 1`] = `
{
  "info": {
    "accessRole": null,
    "experimentId": null,
    "experimentName": null,
    "pipelineVersion": null,
    "sampleIds": [],
  },
  "originalProcessing": {
    "configureEmbedding": {
      "embeddingSettings": {
        "newProperty": {
          "name": "a",
          "value": "b",
        },
      },
    },
  },
  "processing": {
    "configureEmbedding": {
      "embeddingSettings": {
        "newProperty": {
          "name": "a",
          "value": "b",
        },
      },
    },
    "meta": {
      "changedQCFilters": Set {},
      "completingStepError": false,
      "loading": false,
      "loadingSettingsError": false,
    },
  },
}
`;

exports[`experimentSettingsReducer Adds new value properly 1`] = `
{
  "info": {
    "accessRole": null,
    "experimentId": null,
    "experimentName": null,
    "pipelineVersion": null,
    "sampleIds": [],
  },
  "originalProcessing": {
    "configureEmbedding": {
      "embeddingSettings": {
        "newProperty": "property",
      },
    },
  },
  "processing": {
    "configureEmbedding": {
      "embeddingSettings": {
        "newProperty": "property",
      },
    },
    "meta": {
      "changedQCFilters": Set {},
      "completingStepError": false,
      "loading": false,
      "loadingSettingsError": false,
    },
  },
}
`;

exports[`experimentSettingsReducer Changing filter enabled property updates sample filter enabled property as well 1`] = `
{
  "info": {
    "accessRole": null,
    "experimentId": null,
    "experimentName": null,
    "pipelineVersion": 1,
    "sampleIds": [],
  },
  "originalProcessing": {
    "cellSizeDistribution": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 200,
          "minCellSize": 10800,
        },
      },
    },
    "classifier": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "FDR": 0.1,
        },
        "prefiltered": false,
      },
    },
    "configureEmbedding": {
      "clusteringSettings": {
        "method": "louvain",
        "methodSettings": {
          "louvain": {
            "resolution": 0.5,
          },
        },
      },
      "embeddingSettings": {
        "method": "umap",
        "methodSettings": {
          "tsne": {
            "learningRate": 200,
            "perplexity": 30,
          },
          "umap": {
            "distanceMetric": "euclidean",
            "minimumDistance": 0.1,
          },
        },
        "useSaved": false,
      },
      "enabled": true,
    },
    "dataIntegration": {
      "analysisTool": "seurat",
      "dataIntegration": {
        "method": "harmony",
        "methodSettings": {
          "fastmnn": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "harmony": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "seuratv4": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "unisample": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
        },
      },
      "dimensionalityReduction": {
        "excludeGeneCategories": [],
        "method": "rpca",
        "numPCs": 30,
      },
      "enabled": true,
    },
    "doubletScores": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 0.05,
          "probabilityThreshold": 0.2,
        },
      },
    },
    "meta": {
      "changedQCFilters": Set {},
      "completingStepError": false,
      "loading": false,
      "loadingSettingsError": false,
    },
    "mitochondrialContent": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "method": "absoluteThreshold",
          "methodSettings": {
            "absoluteThreshold": {
              "binStep": 200,
              "maxFraction": 0.1,
            },
          },
        },
      },
    },
    "numGenesVsNumUmis": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "regressionType": "gam",
          "regressionTypeSettings": {
            "gam": {
              "p.level": 0.00009,
            },
          },
        },
      },
    },
  },
  "processing": {
    "cellSizeDistribution": {
      "sample-KO": {
        "auto": true,
        "enabled": false,
        "filterSettings": {
          "binStep": 200,
          "minCellSize": 10800,
        },
      },
    },
    "classifier": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "FDR": 0.1,
        },
        "prefiltered": false,
      },
    },
    "configureEmbedding": {
      "clusteringSettings": {
        "method": "louvain",
        "methodSettings": {
          "louvain": {
            "resolution": 0.5,
          },
        },
      },
      "embeddingSettings": {
        "method": "umap",
        "methodSettings": {
          "tsne": {
            "learningRate": 200,
            "perplexity": 30,
          },
          "umap": {
            "distanceMetric": "euclidean",
            "minimumDistance": 0.1,
          },
        },
        "useSaved": false,
      },
      "enabled": true,
    },
    "dataIntegration": {
      "analysisTool": "seurat",
      "dataIntegration": {
        "method": "harmony",
        "methodSettings": {
          "fastmnn": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "harmony": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "seuratv4": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "unisample": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
        },
      },
      "dimensionalityReduction": {
        "excludeGeneCategories": [],
        "method": "rpca",
        "numPCs": 30,
      },
      "enabled": true,
    },
    "doubletScores": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 0.05,
          "probabilityThreshold": 0.2,
        },
      },
    },
    "meta": {
      "changedQCFilters": Set {
        "cellSizeDistribution",
      },
      "completingStepError": false,
      "loading": false,
      "loadingSettingsError": false,
    },
    "mitochondrialContent": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "method": "absoluteThreshold",
          "methodSettings": {
            "absoluteThreshold": {
              "binStep": 200,
              "maxFraction": 0.1,
            },
          },
        },
      },
    },
    "numGenesVsNumUmis": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "regressionType": "gam",
          "regressionTypeSettings": {
            "gam": {
              "p.level": 0.00009,
            },
          },
        },
      },
    },
  },
}
`;

exports[`experimentSettingsReducer Correctly updates pipelineVersion 1`] = `
{
  "info": {
    "accessRole": null,
    "experimentId": null,
    "experimentName": null,
    "pipelineVersion": 2,
    "sampleIds": [],
  },
  "originalProcessing": {
    "cellSizeDistribution": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 200,
          "minCellSize": 10800,
        },
      },
    },
    "classifier": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "FDR": 0.1,
        },
        "prefiltered": false,
      },
    },
    "configureEmbedding": {
      "clusteringSettings": {
        "method": "louvain",
        "methodSettings": {
          "louvain": {
            "resolution": 0.5,
          },
        },
      },
      "embeddingSettings": {
        "method": "umap",
        "methodSettings": {
          "tsne": {
            "learningRate": 200,
            "perplexity": 30,
          },
          "umap": {
            "distanceMetric": "euclidean",
            "minimumDistance": 0.1,
          },
        },
        "useSaved": false,
      },
      "enabled": true,
    },
    "dataIntegration": {
      "analysisTool": "seurat",
      "dataIntegration": {
        "method": "harmony",
        "methodSettings": {
          "fastmnn": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "harmony": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "seuratv4": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "unisample": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
        },
      },
      "dimensionalityReduction": {
        "excludeGeneCategories": [],
        "method": "rpca",
        "numPCs": 30,
      },
      "enabled": true,
    },
    "doubletScores": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 0.05,
          "probabilityThreshold": 0.2,
        },
      },
    },
    "meta": {
      "changedQCFilters": Set {},
      "completingStepError": false,
      "loading": false,
      "loadingSettingsError": false,
    },
    "mitochondrialContent": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "method": "absoluteThreshold",
          "methodSettings": {
            "absoluteThreshold": {
              "binStep": 200,
              "maxFraction": 0.1,
            },
          },
        },
      },
    },
    "numGenesVsNumUmis": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "regressionType": "gam",
          "regressionTypeSettings": {
            "gam": {
              "p.level": 0.00009,
            },
          },
        },
      },
    },
  },
  "processing": {
    "cellSizeDistribution": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 200,
          "minCellSize": 10800,
        },
      },
    },
    "classifier": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "FDR": 0.1,
        },
        "prefiltered": false,
      },
    },
    "configureEmbedding": {
      "clusteringSettings": {
        "method": "louvain",
        "methodSettings": {
          "louvain": {
            "resolution": 0.5,
          },
        },
      },
      "embeddingSettings": {
        "method": "umap",
        "methodSettings": {
          "tsne": {
            "learningRate": 200,
            "perplexity": 30,
          },
          "umap": {
            "distanceMetric": "euclidean",
            "minimumDistance": 0.1,
          },
        },
        "useSaved": false,
      },
      "enabled": true,
    },
    "dataIntegration": {
      "analysisTool": "seurat",
      "dataIntegration": {
        "method": "harmony",
        "methodSettings": {
          "fastmnn": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "harmony": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "seuratv4": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "unisample": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
        },
      },
      "dimensionalityReduction": {
        "excludeGeneCategories": [],
        "method": "rpca",
        "numPCs": 30,
      },
      "enabled": true,
    },
    "doubletScores": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 0.05,
          "probabilityThreshold": 0.2,
        },
      },
    },
    "meta": {
      "changedQCFilters": Set {},
      "completingStepError": false,
      "loading": false,
      "loadingSettingsError": false,
    },
    "mitochondrialContent": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "method": "absoluteThreshold",
          "methodSettings": {
            "absoluteThreshold": {
              "binStep": 200,
              "maxFraction": 0.1,
            },
          },
        },
      },
    },
    "numGenesVsNumUmis": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "regressionType": "gam",
          "regressionTypeSettings": {
            "gam": {
              "p.level": 0.00009,
            },
          },
        },
      },
    },
  },
}
`;

exports[`experimentSettingsReducer Updates existing value properly 1`] = `
{
  "info": {
    "accessRole": null,
    "experimentId": null,
    "experimentName": null,
    "pipelineVersion": null,
    "sampleIds": [],
  },
  "originalProcessing": {
    "configureEmbedding": {
      "embeddingSettings": {
        "method": "newMethod",
      },
    },
  },
  "processing": {
    "configureEmbedding": {
      "embeddingSettings": {
        "method": "newMethod",
      },
    },
    "meta": {
      "changedQCFilters": Set {},
      "completingStepError": false,
      "loading": false,
      "loadingSettingsError": false,
    },
  },
}
`;

exports[`experimentSettingsReducer Updates sample settings properly 1`] = `
{
  "info": {
    "accessRole": null,
    "experimentId": null,
    "experimentName": null,
    "pipelineVersion": 1,
    "sampleIds": [],
  },
  "originalProcessing": {
    "cellSizeDistribution": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 200,
          "minCellSize": 10800,
        },
      },
    },
    "classifier": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "FDR": 0.1,
        },
        "prefiltered": false,
      },
    },
    "configureEmbedding": {
      "clusteringSettings": {
        "method": "louvain",
        "methodSettings": {
          "louvain": {
            "resolution": 0.5,
          },
        },
      },
      "embeddingSettings": {
        "method": "umap",
        "methodSettings": {
          "tsne": {
            "learningRate": 200,
            "perplexity": 30,
          },
          "umap": {
            "distanceMetric": "euclidean",
            "minimumDistance": 0.1,
          },
        },
        "useSaved": false,
      },
      "enabled": true,
    },
    "dataIntegration": {
      "analysisTool": "seurat",
      "dataIntegration": {
        "method": "harmony",
        "methodSettings": {
          "fastmnn": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "harmony": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "seuratv4": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "unisample": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
        },
      },
      "dimensionalityReduction": {
        "excludeGeneCategories": [],
        "method": "rpca",
        "numPCs": 30,
      },
      "enabled": true,
    },
    "doubletScores": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 0.05,
          "probabilityThreshold": 0.2,
        },
      },
    },
    "meta": {
      "changedQCFilters": Set {},
      "completingStepError": false,
      "loading": false,
      "loadingSettingsError": false,
    },
    "mitochondrialContent": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "method": "absoluteThreshold",
          "methodSettings": {
            "absoluteThreshold": {
              "binStep": 200,
              "maxFraction": 0.1,
            },
          },
        },
      },
    },
    "numGenesVsNumUmis": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "regressionType": "gam",
          "regressionTypeSettings": {
            "gam": {
              "p.level": 0.00009,
            },
          },
        },
      },
    },
  },
  "processing": {
    "cellSizeDistribution": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 400,
          "minCellSize": 10800,
        },
      },
    },
    "classifier": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "FDR": 0.1,
        },
        "prefiltered": false,
      },
    },
    "configureEmbedding": {
      "clusteringSettings": {
        "method": "louvain",
        "methodSettings": {
          "louvain": {
            "resolution": 0.5,
          },
        },
      },
      "embeddingSettings": {
        "method": "umap",
        "methodSettings": {
          "tsne": {
            "learningRate": 200,
            "perplexity": 30,
          },
          "umap": {
            "distanceMetric": "euclidean",
            "minimumDistance": 0.1,
          },
        },
        "useSaved": false,
      },
      "enabled": true,
    },
    "dataIntegration": {
      "analysisTool": "seurat",
      "dataIntegration": {
        "method": "harmony",
        "methodSettings": {
          "fastmnn": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "harmony": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "seuratv4": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
          "unisample": {
            "normalisation": "logNormalize",
            "numGenes": 2000,
          },
        },
      },
      "dimensionalityReduction": {
        "excludeGeneCategories": [],
        "method": "rpca",
        "numPCs": 30,
      },
      "enabled": true,
    },
    "doubletScores": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "binStep": 0.05,
          "probabilityThreshold": 0.2,
        },
      },
    },
    "meta": {
      "changedQCFilters": Set {},
      "completingStepError": false,
      "loading": false,
      "loadingSettingsError": false,
    },
    "mitochondrialContent": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "method": "absoluteThreshold",
          "methodSettings": {
            "absoluteThreshold": {
              "binStep": 200,
              "maxFraction": 0.1,
            },
          },
        },
      },
    },
    "numGenesVsNumUmis": {
      "sample-KO": {
        "auto": true,
        "enabled": true,
        "filterSettings": {
          "regressionType": "gam",
          "regressionTypeSettings": {
            "gam": {
              "p.level": 0.00009,
            },
          },
        },
      },
    },
  },
}
`;
